### DISCUSSION TITLE
Copliot now requires Node installed locally

### DISCUSSION THREAD

**Discussion by aneroid** [community/community#135791] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

On PyCharm Pro, the GitHub Copilot plugin got updated to version `1.5.18.6517`. But it now shows the error:

> **GitHub Copilot Unavailable**
>
> Unfortunately, GitHub Copilot is unsupported on this machine. Please make sure that Node.js is installed and that "node" is available in the PATH."

![image](https://github.com/user-attachments/assets/b0aaed42-32d6-46f5-88b1-83bdee8d2416)

Since when was Node a requirement for the plugin?

On Intellij IDEA, it's at `1.5.17.6356` (older) and it's working fine.

Are they really expecting all developers to have Node.js installed _locally_? And which version? If I need Node for something, I only use it via Docker. So at least allow alternative environments (Docker, WSL2, etc.) to be configured.

Edit: After a couple of restarts, the Copilot plugin on PyCharm can't login anymore: _"Failed to initiate the GitHub login process. Please try again."_

Aside:
- I presume  `1.5.17...` < `1.5.18...` (if it's Semantic versioning)
- And if the last part implies `...6517` < `...6356` - I've tried "checking for updates" multiple times and there's nothing newer showing up on PyCharm

**Comment by tFerrellBastian** [comment]
I had a similar issue, Running Pycharm pro latest version and updating to Github CoPilot v[1.5.18.6517](https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable/586950). I do have node installed locally and it is in my PATH variable. It looks like 1.5.18 has a bug that breaks it. (system is ubuntu 22) 

While they work on a fix, I recommend manually downgrading to 1.5.17 which you can do by downloading [here](https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable) from jet brains and install using a local package install until they have a fix for it.

**Comment by jgcmarins** [comment]
I just updated WebStorm and saw the same problem.
```
WebStorm 2024.2
Build #WS-242.20224.342, built on August 9, 2024
Runtime version: 21.0.3+13-b509.4 aarch64 (JCEF 122.1.9)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.lwawt.macosx.LWCToolkit
macOS 14.5
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 4096M
Cores: 10
Metal Rendering is ON
Registry:
  ide.experimental.ui=true
  typescript.compiler.evaluation.fallback=true
Non-Bundled Plugins:
  com.wakatime.intellij.plugin (15.0.2)
  com.intellij.kubernetes (242.20224.175)
  com.intellij.ml.llm (242.20224.393)
  com.github.copilot (1.5.18.6517)
  mdx.js (242.20224.159)
```

**Comment by thimoti26** [comment]
Hi, I have another problem using PhpStorm and updating Copilot 1.5.18.6517, message is `Failed to initiate the GitHub login process. Please try again.`.

**Comment by AntoineGS** [comment]
+1 with the node.js & login issues 😭
Copilot running on the Client with PhpStorm and WSL with NodeJS 20 installed.

**Comment by AmiDavidW** [comment]
Have the same error with IntelliJ IDEA 2024.2 and Copilot 1.5.18.6517 on Ubuntu 2204.03. 
NodeJS 20 installed

**Comment by johnraz** [comment]
This seems to only happen on the nightly build, latest stable version 1.5.17.6356 works fine.

**Comment by Akash1134** [comment, marked as answer]
👋 Hi everyone, 

We wanted to provide an update regarding the issue some of you have reported with version 1.5.18. **We recommend reinstalling the plugin, which will revert you back to  v1.5.17**. This should help restore functionality while we work on a fix. We appreciate your patience and understanding as we address this.

Marking this as the answer for visibility

Thanks!

**Reply by thomas-sickert** [reply]
Update: `1.5.19` is out with a fix if you didn't revert to `1.5.17`. Please let us know if you see any additional issues.

**Comment by MysterHawk** [comment]
Now it works for me! Thx

**Comment by greew** [comment]
Just upgraded to 1.5.19.6527 and it works!

**Comment by collado-mike** [comment]
I still have this problem in Intellij 2024.2 with Copilot 1.5.20.6554 installed.

**Reply by collado-mike** [reply]
Nvm. This is for a remote Intellij backend. I installed the plugin on the client rather than the host and that works.

**Comment by darshansarojini** [comment]
I have the same bug:
Unfortunately, GitHub Copilot is unsupported on this machine. Please make sure that Node.js is installed and that "node" is available in the PATH.
I am running Windows on ARM on Microsoft Surface Laptop 7 with Snapdragon X Elite
GitHub Copilot 1.5.29.7524

**Reply by adamswann** [reply]
I was receiving the same message on a Surface Laptop 7 with Snapdragon X Plus and a clean Windows on ARM install.  For me, installing `node` and making sure it was in the PATH indeed resolved it.

```
winget install -e --id CoreyButler.NVMforWindows
nvm install 22.12.0
nvm use 22.12.0
```

Then restarted Rider, Copilot asked for GitHub authorization, and now it appears to work.

**Reply by ramonsb123** [reply]
adamswann solution worked for me too :) Same setup. I had tried installing Node on WSL but it didn't work.

**Reply by darshansarojini** [reply]
Thank you. Installing nodejs fixed this issue.

**Reply by BenjaminDemouge** [reply]
same issue, is there a way to make copilot work without installing node in the path ?

**Comment by AnasIbrahim** [comment]
I had the same problem running copilot on ARM cpu. The problem was nodejs not installed. Also nodejs 18 was required while default was 10. Installing nodejs 18 just fixed the problem.

**Comment by hrishikesh-k** [comment]
So based on the recent comments, this doesn't appear to be fixed, right? I am also using Windows ARM. I mainly do all my development work in WSL. Node.js is also installed in WSL. Like the last few comments say, I can likely install Node.js on Windows, but since when is this a requirement?

**Comment by Frost-Mactavish** [comment]
run into the same problem on PyCharm Pro 2025.1 after updating the plugin to 1.5.48-243

**Reply by Frost-Mactavish** [reply]
fixed after reinstalling the plugin, what a mess

**Comment by automatix** [comment]
Hi guys,

I'm observing this issue in the `v1.5.57-243`:

<img width="1355" height="737" alt="image" src="https://github.com/user-attachments/assets/d99bb754-2228-4014-bfd8-bc8153af8d1a" />

**Comment by tip2tail** [comment]
Randomly started getting this error in PHPStorm now, is this a regression?

**Comment by Alintya** [comment]
Also seeing this on an arm64 windows machine.
